artemis-server/src/main/resources/schema/artemis-configuration.xsd

```diff
@@ -16,18 +16,23 @@
   limitations under the License.
 -->
 <xsd:schema xmlns:xsd="http://www.w3.org/2001/XMLSchema"
+            xmlns:xi="http://www.w3.org/2001/XInclude"
             xmlns="urn:activemq:core"
             targetNamespace="urn:activemq:core"
             attributeFormDefault="unqualified"
             elementFormDefault="qualified"
             version="1.0">
 
    <xsd:import namespace="http://www.w3.org/XML/1998/namespace" schemaLocation="xml.xsd"/>
+   <xsd:import namespace="http://www.w3.org/2001/XInclude" schemaLocation="XInclude.xsd"/>
 
    <xsd:element name="core" type="configurationType"/>
 
    <xsd:complexType name="configurationType">
       <xsd:all>
+
+         <xsd:element ref="xi:include" maxOccurs="1" minOccurs="0"/>
```

Comment:
This is not needed Includes already works

Reply:
{{<xi:include>}} is not currently a valid element in {{configurationType}}, leading syntax validation tooling to balk. Including this element ref is the reason for adding the XInclude.xsd file.

XSD 1.0 only allows {{all}} groups to have {{minOccurs == 0}} and {{maxOccurs == 1}}, while valid configurations of course could have multiple included element files. XSD 1.1 drops this restriction but including an XSD 1.1 validator turned out to be exceedingly painful.

Reply:
Against the jira thats linked here the need is to have divert in an external config file imported.... this is possible today without these mass changes